tests/model/documentselection.js

```diff
@@ -1778,6 +1778,63 @@ describe( 'DocumentSelection', () => {
 				// Now it's clear that it's the default range.
 				expect( selection.getFirstPosition().path ).to.deep.equal( [ 0, 0 ] );
 			} );
+
+			it( 'does not break if multi-range selection is inside text nodes', () => {
```

Comment:
"Does not break" does not explain what should actually happen. I prefer explaining what's the expected result.

```suggestion
			it( 'handles multi-range selection in a text node by merging it into one range', () => {
```

Reply:
BTW, as mentioned in the test description – I'd actually expect the ranges to be merged if they are identical. Would that require a lot of work?

Reply:
> I'd actually expect the ranges to be merged if they are identical.

I don't see a need to merging two identical ranges. With intersecting ones I would  have to check.